cloudsync/tasks.py

```diff
@@ -123,7 +126,63 @@ def transcode_from_s3(self, video_id):  # pylint: disable=unused-argument
     task_id = self.get_task_id()
     self.update_state(task_id=task_id, state=VideoStatus.TRANSCODING)
 
-    video_file = VideoFile.objects.get(video__id=video_id, encoding='original')
+    video_file = video.videofile_set.get(encoding='original')
+
+    try:
+        transcode_video(video, video_file)
+    except ClientError:
+        self.update_state(task_id=task_id, state=states.FAILURE)
+        raise
+
+
+@shared_task(bind=True)
+def update_video_statuses(self):  # pylint: disable=unused-argument
+    """
+    Check on statuses of all transcoding videos and update their status if appropriate
+    """
+    transcoding_videos = Video.objects.filter(status=VideoStatus.TRANSCODING)
+    for video in transcoding_videos:
+        try:
+            refresh_status(video)
+        except EncodeJob.DoesNotExist:
+            # Log the exception but don't raise it so other videos can be checked.
+            log.exception("No EncodeJob object exists for video id %d", video.id)
+            video.update_status(VideoStatus.TRANSCODE_FAILED)
+        except ClientError as exc:
+            # Log the exception but don't raise it so other videos can be checked.
+            log.exception("AWS error when refreshing job status for video %d: %s", video.id, exc.response)
+            video.update_status(VideoStatus.TRANSCODE_FAILED)
+
+
+@shared_task(bind=True)
+def monitor_watch_bucket(self):  # pylint: disable=unused-argument
+    """
+    Check the watch bucket for any files and import them if found
+    """
+    watch_bucket = get_bucket(settings.VIDEO_S3_WATCH_BUCKET)
+    error = None
+    for key in watch_bucket.objects.all():
+        try:
+            process_watch_file(key.key)
+        except (ClientError, Exception, ) as exc:  # pylint: disable=broad-except
+            # Log any exception, raise later so other files can be processed.
+            msg = exc.response if isinstance(exc, ClientError) else str(exc)
+            log.exception("AWS error when ingesting file from watch bucket %s: %s", key.key, msg)
+            error = exc
+    if isinstance(error, Exception):
+        # Raise the last exception
+        raise error   # pylint: disable=raising-bad-type
```

Comment:
I am not raising the error before, just logging it, so that an exception with one file won't prevent other files from being processed.  Then the last error logged (if any) will be raised once all files have been tried.